File: settlement/room/infra/importt/Crate.java
package settlement.room.infra.importt;

import settlement.room.main.RoomInstance;
import settlement.room.main.job.StorageCrate;

final class Crate extends StorageCrate{

	protected final ROOM_IMPORT b;
	ImportInstance ins;
	
	Crate(ROOM_IMPORT b) {
		this.b = b;
	}

	@Override
	protected boolean is(int tx, int ty) {
		if (b.is(tx, ty)) {
			ins = b.getter.get(tx, ty);
			if (b.constructor.isCrate(tx, ty)) {
				return true;
			}
		}
		return false;
	}
	
	@Override
	protected void count(int delta) {
		ins.count(this, delta);
	}
	
	@Override
	protected int max(RoomInstance ins) {
		return ImportInstance.crateMax;
	}

	@Override
	protected double spoilRate(RoomInstance ins) {
		return 1.0;
	}
	
	@Override
	public boolean isStorage() {
		return false;
	}
	
	@Override
	public boolean isPrio() {
		return false;
	}
	
	@Override
	public boolean storageIsFindable() {
		return false;
	}
	
}

File: LegacySource/settlement/room/infra/importt/Crate.cs
using settlement.room.main;
using settlement.room.main.job;

namespace settlement.room.infra.importt
{
    internal sealed class Crate : StorageCrate
    {
        private readonly ROOM_IMPORT b;
        private ImportInstance ins;

        public Crate(ROOM_IMPORT b)
        {
            this.b = b;
        }

        protected override bool Is(int tx, int ty)
        {
            if (b.Is(tx, ty))
            {
                ins = b.Getter.Get(tx, ty);
                if (b.Constructor.IsCrate(tx, ty))
                {
                    return true;
                }
            }
            return false;
        }

        protected override void Count(int delta)
        {
            ins.Count(this, delta);
        }

        protected override int Max(RoomInstance ins)
        {
            return ImportInstance.CrateMax;
        }

        protected override double SpoilRate(RoomInstance ins)
        {
            return 1.0;
        }

        public override bool IsStorage()
        {
            return false;
        }

        public override bool IsPrio()
        {
            return false;
        }

        public override bool StorageIsFindable()
        {
            return false;
        }
    }
}